src/main/java/com/dnastack/ddap/explore/common/ReactiveDamOAuthFacadeClient.java

```diff
@@ -0,0 +1,95 @@
+package com.dnastack.ddap.explore.common;
+
+import com.dnastack.ddap.common.client.WebClientFactory;
+import com.dnastack.ddap.explore.common.config.DamFacadeConfig;
+import com.dnastack.ddap.explore.dam.client.ReactiveDamOAuthClient;
+import com.dnastack.ddap.ic.oauth.model.TokenResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.util.List;
+
+public class ReactiveDamOAuthFacadeClient implements ReactiveDamOAuthClient {
+    private final DamFacadeConfig damFacadeConfig;
+
+    public ReactiveDamOAuthFacadeClient(DamFacadeConfig damFacadeConfig) {
+        this.damFacadeConfig = damFacadeConfig;
+    }
+
+    @Override
+    public URI getAuthorizeUrl(String realm, String state, String scopes, URI redirectUri, List<URI> resources) {
+        var builder = UriComponentsBuilder.fromUri(URI.create(damFacadeConfig.getBaseUrl() + "/api/v1alpha/standalone-mode/oauth2/auth"))
+                .queryParam("response_type", "code")
+                .queryParam("client_id", damFacadeConfig.getClientId())
+                .queryParam("redirect_uri", redirectUri)
+                .queryParam("resource", resources.toArray())
+                .queryParam("state", state);
+
+        if (scopes != null) {
+            builder.queryParam("scope", scopes);
+        }
+
+        return builder.build().toUri();
+    }
+
+    @Override
+    public URI getLegacyAuthorizeUrl(String realm, String state, String scopes, URI redirectUri, List<URI> resources) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Mono<TokenResponse> exchangeAuthorizationCodeForTokens(String realm, URI redirectUri, String code) {
+        return Mono.just(TokenResponse.builder()
+                .accessToken("fake access token")
+                .idToken("fake id token - not used")
+                .refreshToken("fake refresh token")
+                .build());
+    }
+
+    @Override
+    public Mono<TokenResponse> legacyExchangeAuthorizationCodeForTokens(String realm, URI redirectUri, String code) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Mono<HttpStatus> testAuthorizeEndpoint(URI uri) {
+        return WebClientFactory.getWebClient()
+                .get()
+                .uri(uri)
+                .exchange()
+                .map(ClientResponse::statusCode);
+    }
+
+    @Override
+    public Mono<TokenResponse> refreshAccessToken(String realm, String refreshToken, String scope) {
+        throw new UnsupportedOperationException();
```

Comment:
Not necessary now, but we should add an exception handler for this error that returns a `DdapErrorResponse` with 501 unimplemented.